### PR TITLE
fix(check-status): enhance response status and add abstraction document public

### PR DESCRIPTION
**Pull request by indraprasetya154**
## Overview
New request from elogistik integration

fix(check-status): enhance response status and add abstraction document public
- add signed response (boolean)
- add registered response (boolean)
- add public url but with abstraction via controller

## Evidence:
project: SIDEBAR
title: fix(check-status): enhance response status and add abstraction document public
participants: @indraprasetya154 @azophy @fajarhikmal214

### CODE REVIEW

File: src/app/Http/Controllers/V1/EsignDocumentCheckStatusController.php [hunk 1]
```diff
@@ -21,20 +21,22 @@ public function __invoke(Request $request)
...
         });
+
         $result = collect($tids)->map(fn($tid) =>
             [
                 'tid' => $tid,
                 'is_exist' => !empty($documents_by_tid[$tid]),
-                'data' => !empty($documents_by_tid[$tid]) ?
-                          $documents_by_tid[$tid] :
+                'signed' => $documents_by_tid[$tid]->status,
+                'registered' => $documents_by_tid[$tid]->is_registered,
```
Comment (azophy) [inline src/app/Http/Controllers/V1/EsignDocumentCheckStatusController.php:32]: is the result from this 2 fields always in boolean (true/false) or could it be in other form (in integer) ? when you show me the results of this endpoint, I see that the results form these fields are in integer. I think its better if we could provide it in explicit boolean values. CMIIW

I also think that the field name should be `is_signed` & `is_registered` just like `is_exist`. however this is only if the field names haven't been agreed in previous discussion with dinkes. if its already been agreed before, than its better to stick to existing agreement.

Comment (indraprasetya154) [reply src/app/Http/Controllers/V1/EsignDocumentCheckStatusController.php:32]: aaah my bad, its integer and not boolean, it will be fixed later

Comment (indraprasetya154) [reply src/app/Http/Controllers/V1/EsignDocumentCheckStatusController.php]: already fixed mas [94c4d5f](https://github.com/jabardigitalservice/office-services/pull/676/commits/94c4d5fa57551515ea764e8892d086789313c1df)

the filed didn't have any agreement with stakeholders, so its flexible

**Review by azophy — APPROVED** [review]
nice 👍

**Review by fajarhikmal214 — APPROVED** [review]

**Pull request merged**